mkosi/__init__.py

```diff
@@ -1286,14 +1287,29 @@ def gzip_binary(context: Context) -> str:
     return "pigz" if context.config.find_binary("pigz") else "gzip"
 
 
+def _kernel_get_ver_from_modules(context: Context) -> str:
+    moddir = context.root / "usr/lib/modules"
+    # try to get version from the first dir under usr/lib/modules
+
+    for item in moddir.glob("[0-9]*"):
+        if item.is_dir() and re.match(r'\d+\.\d+.*', item.name):
+            return item.name
```

Comment:
This might be an edge case, but I'm not sure glob guarantees an order, so how about
```suggestion
    kvers = sorted(
        [
            item.name
            for item in moddir.glob("[0-9]*")
            if item.is_dir() and re.match(r"\d+\.\d+.*", item.name)
        ],
        key=lambda p: GenericVersion(p)
    )
```
and then returning the zeroeth element, if there is one?